Rocket.Chat/Views/Cells/Chat/ChatMessageImageView.swift

```diff
@@ -43,22 +46,64 @@ final class ChatMessageImageView: UIView {
         return UITapGestureRecognizer(target: self, action: #selector(didTapView))
     }()
 
+    private func checkHttpImage(imageURL: URL) {
+        if imageURL.absoluteString.starts(with: "http://") {
+            isLoadable = false
+            labelTitle.text = attachment.title + " (" + localized("alert.insecure_image.title") + ")"
+            placeholderImage = UIImage(named: "Resource Unavailable")
+            imageView.contentMode = UIViewContentMode.center
+        }
+    }/*
+
+    private func openImageExternal(imageURL: URL) {
+        let controller = SFSafariViewController(url: imageURL)
+        present(controller, animated: true)
+    }*/
+
     fileprivate func updateMessageInformation() {
         let containsGesture = gestureRecognizers?.contains(tapGesture) ?? false
         if !containsGesture {
             addGestureRecognizer(tapGesture)
         }
 
+        guard let imageURL = attachment.fullImageURL() else {
+            return
+        }
+
         labelTitle.text = attachment.title
+        checkHttpImage(imageURL: imageURL)
 
-        let imageURL = attachment.fullImageURL()
         activityIndicatorImageView.startAnimating()
-        imageView.sd_setImage(with: imageURL, completed: { [weak self] _, _, _, _ in
+        imageView.sd_setImage(with: imageURL, placeholderImage: placeholderImage, completed: { [weak self] _, _, _, _ in
```

Comment:
Why do you think we need to try loading the HTTP image if we know (before on the `checkHttpImage`) that we don't want to? Maybe we could make a check and don't even try to load it. What do you think? You could also remove the instance variable `placeholderImage` and set direct the image of `imageView`.